test/TestProjects/MgmtScopeResource/Generated/PolicyAssignmentContainer.cs

```diff
@@ -270,207 +269,343 @@ public async virtual Task<Response<bool>> CheckIfExistsAsync(string policyAssign
-                        else
-                        {
-                            var parent = Id.Parent;
-                            var parentParts = new List<string>();
-                            while (!parent.ResourceType.Equals(ResourceGroup.ResourceType))
-                            {
-                                parentParts.Insert(0, $"{parent.ResourceType.Types[parent.ResourceType.Types.Count - 1]}/{parent.Name}");
-                                parent = parent.Parent;
-                            }
-                            var parentResourcePath = parentParts.Count > 0 ? string.Join("/", parentParts) : "";
-                            Id.TryGetSubscriptionId(out var subscriptionId);
-                            var response = _restClient.GetForResource(subscriptionId, resourceGroupName, Id.ResourceType.Namespace, parentResourcePath, Id.ResourceType.Types[Id.ResourceType.Types.Count - 1], Id.Name, filter, top, cancellationToken: cancellationToken);
-                            return Page.FromValues(response.Value.Value.Select(value => new PolicyAssignment(Parent, value)), response.Value.NextLink, response.GetRawResponse());
-                        }
-                    }
-                    else if (Id.TryGetSubscriptionId(out _))
-                    {
-                        var response = _restClient.GetAll(Id.Name, filter, top, cancellationToken: cancellationToken);
+                    using var scope = _clientDiagnostics.CreateScope("PolicyAssignmentContainer.GetAll");
+                    scope.Start();
+                    try
+                    {
+                        var response = _policyAssignmentsRestClient.GetForResourceGroupNextPage(nextLink, Id.ResourceGroupName, filter, top, cancellationToken: cancellationToken);
                         return Page.FromValues(response.Value.Value.Select(value => new PolicyAssignment(Parent, value)), response.Value.NextLink, response.GetRawResponse());
                     }
-                    else
+                    catch (Exception e)
+                    {
+                        scope.Failed(e);
+                        throw;
+                    }
+                }
+                return PageableHelpers.CreateEnumerable(FirstPageFunc, NextPageFunc);
+            }
+            else if (Id.ResourceType == "Microsoft.ManagementGroups/managementGroups")
+            {
+                Page<PolicyAssignment> FirstPageFunc(int? pageSizeHint)
+                {
+                    using var scope = _clientDiagnostics.CreateScope("PolicyAssignmentContainer.GetAll");
+                    scope.Start();
+                    try
                     {
-                        var response = _restClient.GetForManagementGroup(Id.Name, filter, top, cancellationToken: cancellationToken);
+                        var response = _policyAssignmentsRestClient.GetForManagementGroup(Id.Parent.Name, filter, top, cancellationToken: cancellationToken);
```

Comment:
The first argument becomes `Id.Parent.Name` which seems wrong.

Reply:
It looks like the issue is caused by subscription parameter is now in the constructor of PolicyAssignmentsRestOperations as you commented UpdateSubscriptionIdForTenantIdResource in MgmtOutputLibrary.